### ISSUE TITLE
"Package not found" error when "--force" is used

### ISSUE THREAD

**Issue by toabctl** [openSUSE/zypper#289]
This is with zypper 1.14.27-1.11 on a SLE15SP1 machine.

I have 2 repos (called `cloud` and `cloudupstream`). The `cloud` repo contains the package `opentack-quickstart`:
```
# zypper info openstack-quickstart
...
Repository     : cloud                                                     
Name           : openstack-quickstart
...
```

The `cloudupstream` repo contains the package `openstack-tempest` (which provides `openstack-tempest-test`).:
```
# zypper info --provides openstack-tempest
...
Repository     : upstreamcloud                                 
Name           : openstack-tempest
...
Provides       : [3]                                           
    config(openstack-tempest) = 21.0.0-4.3
    openstack-tempest = 21.0.0-4.3
    openstack-tempest-test = 21.0.0-4.3
```

When I try to install both packages now with the `--force` option, the installation fails:
```
# zypper --non-interactive -n install --force openstack-quickstart openstack-tempest-test
Warning: The flag --non-interactive can only be used once.
Loading repository data...
Reading installed packages...
Forcing installation of 'openstack-quickstart-2019.2+git.1567757686.5170c66-1.1.noarch' from repository 'cloud'.
Package 'openstack-tempest-test' not found.
```

I expected that the installation suceeds.

When I skip the `--force` option, it works:
```
# zypper --non-interactive -n install openstack-quickstart openstack-tempest-test
Warning: The flag --non-interactive can only be used once.
Loading repository data...
Reading installed packages...

'openstack-tempest-test' not found in package names. Trying capabilities.
Resolving package dependencies...

The following 8 NEW packages are going to be installed:
  crudini openstack-quickstart openstack-tempest python2-iniparse python3-future python3-python-subunit python3-stestr python3-tempest
...
```

**Comment by bzeller** [comment]
Fixed in zypper > 1.14.30

**Comment by bzeller** [comment]
There is one limitation, reinstalling a already installed package works only by providing the package name. This is due to limitation in the current libzypp API.
